libraries/WiFi/src/SecSocket.h

```diff
@@ -41,7 +41,7 @@ class Socket {
 
     void listen(int max_connections);
     bool accept(Socket & client_socket);
-    uint32_t send(const void *data, uint32_t len);
+    uint32_t send(const void *data, uint32_t len, cy_socket_sockaddr_t *addr = NULL);
```

Comment:
In order to keep this an Arduino level API, i would use here an IPAddress object as argument. That way we don´t need to handle any cyhal details when using the Socket class. 
Like here:
https://github.com/Infineon/arduino-core-psoc6/blob/main/libraries/WiFi/src/SecSocket.cpp#L81-L88